### PR TITLE
clean_olid(): Deal with olid == None

### LINKED ISSUE #46: Submitting recommendation fails with a 500 error
While testing recommendation submissions, a submission failed due to `lower()` being called on a null object in `Book.clean_olid()`.

`clean_olid` is called on each candidate book in a recommendation.

### PR DESCRIPTION
**Pull request by cclauss**
Fixes #46

@jamesachamp

### CODE REVIEW

File: bestbook/api/books.py [hunk 1]
```diff
@@ -70,6 +70,7 @@ class Book(core.Base):
     @staticmethod
     def clean_olid(olid):
         """Extract just the olid from some olid-containing string"""
+        olid = olid or ''
         if olid.lower().startswith('ol') and olid.lower()[-1] in ['M', 'W']:
             return olid
         return re.findall(r'OL[0-9]+[MW]', olid)[0]
```
Comment (jamesachamp) [inline bestbook/api/books.py:76]: I think that this will throw an IndexError if `olid` is an empty string.

**Review by jamesachamp — COMMENTED** [review]
When I last worked on the recommendation submission flow, I removed some front-end validations.  Reinstating these validations will solve this issue, and I'll work on this today.

I'm of the mind that we should also be validating all submissions on the server as well.  What are your thoughts on this?

File: bestbook/api/books.py (hunk 1 above)
Comment (cclauss) [reply bestbook/api/books.py]: Good catch.  Fixed.

**Comment by cclauss** [comment]
Does this function do the right thing if the olid is `OL123M, OL789W` ?

**Comment by jamesachamp** [comment]
It does, but I'm now seeing that the second if statement always returns false.  The last character of a lowercase OLID will never be 'M' or 'W'.

File: bestbook/api/books.py [hunk 2]
```diff
@@ -69,7 +69,16 @@ class Book(core.Base):
...
+        Extract just the olid from some olid-containing string
+        
+        >>> Book.clean_olid(None)
+        ''
+        >>> Book.clean_olid('')
+        ''
+        """
+        if not olid:
+            return ''
         if olid.lower().startswith('ol') and olid.lower()[-1] in ['M', 'W']:
```
Comment (jamesachamp) [inline bestbook/api/books.py:82]:
``` suggestion
        if olid.lower().startswith('ol') and olid[-1] in ['M', 'W']:
```

Comment (cclauss) [reply bestbook/api/books.py:82]: When I look at OLIDs on Open Library, they are uppercase, not lowercase.  So perhaps the logic should be:
```python
if not olid:
    return ''
olid = olid.upper()
[ ... ]
```

Comment (jamesachamp) [reply bestbook/api/books.py]: I think that's great!  Then, all of the persisted values would be formatted consistently.

**Comment by cclauss** [comment]
@jamesachamp Can I please get your review of the latest changes?

**Review by jamesachamp — APPROVED** [review]
Lgtm, thanks @cclauss!

**Review by mekarpeles — APPROVED** [review]

**Pull request merged**